Update to pip==19

tfmorris opened the pull request:
Fixes #2910

Allows us to build Docker images again as a stopgap until with update to Bionic.

@hornc @mekarpeles

cdrini commented on docker/Dockerfile.olbase:
```diff
@@ -29,6 +29,9 @@ RUN apt-get -qq update && apt-get install -y \
     python-yaml
 
+# Update pip to at least 9.0
+RUN pip install --upgrade pip==9
```
The version on ol-dev0 and ol-web3 is 19.3.1

```suggestion
RUN pip install --upgrade pip==19.3.1
```

cclauss replied:
Why not `pip==20.0.1`?

cdrini replied:
That's what's on our production environment. We don't have a great way of updating things there yet (they're not running docker, so we have to update each dep manually, which is a time sink for a 2.5 person team). I think keeping things in sync is more beneficial than keeping things bleeding edge; once we have docker in more places, updating dependencies will be much easier.

tfmorris commented:
Seems like someone who had access to the secret production environment would be in a better position that me to make sure these track, but I've made the requested change.

pip 19 appears to be stricter about uninstalling packages that it didn't install, so I removed the apt-get packages that get overwritten later by pip.

I also fixed a couple of other build errors related to file permissions and missing home directories (e.g. `npm update` didn't work).

Adding `node_modules` to `.dockerignore` is a performance optimization (since I had to iterate so many times on the build).

tfmorris commented:
I'm not super motivated to investigate in detail, but when I checked a week ago, `zipp` had two major version bumps in January, which I consider a smoking gun. It's likely that the new release is incompatible with the ancient version of pip that we're currently using. It's also possible that some other dependency update triggered this transient dependency issue.

If you read the conversation in the issue that I linked to the original problem report, you can see that the incompatibility has been known for a while and the zipp team has no plans to address it when a simple pip upgrade fixes the issue.

cdrini commented on docker/Dockerfile.olbase:
```diff
@@ -43,20 +43,20 @@ RUN ln -s /openlibrary/conf/nginx/sites-available/openlibrary.conf /etc/nginx/si
-USER openlibrary
 RUN ln -s vendor/infogami/infogami infogami
 # run make to initialize git submodules, build css and js files
+RUN npm update caniuse-lite browserslist
```
Ah, looks like something even odder is going on; the deps _do_ look outdated, but calling update doesn't edit the `package-lock.json`? Going to attribute this to our old version of node/npm, so should be addressed in the other PR, but removed from here.

cdrini replied:
Done

cdrini commented:
Darn! Messed that up; one sec and I'll fix it.

cdrini commented on docker/Dockerfile.olbase:
```diff
@@ -7,7 +7,7 @@ ENV LC_ALL POSIX
 # add openlibrary users
 RUN groupadd --system openlibrary \
-  && useradd --no-log-init --system --gid openlibrary openlibrary \
+  && useradd --no-log-init --system --gid openlibrary --create-home --home /openlibrary openlibrary \
```
Done

cdrini commented:
It was easier to just make the requisite changes than to pull out the pip update.

Tested:
- ✅ Doing a full rebuild (after even deleting the xenial image) doesn't error
  - It did show the caniuselist warning, but as noted I think that's a node/npm version issue.
- ✅ Resulting image loads the site
- ✅ `docker-compose exec web test` works

tfmorris commented:
@cdrini Is that working the way you want? You changed the home directory, but none of the other commands that rely on it, so I'm suspicious. For example, won't the node installation now go to `/home/openlibrary` instead of `/openlibrary` as a result of the USER command?
